### PR TITLE
Implement login & integrate phone input

### PR DESCRIPTION
**Pull request by tung2744**
ref #3577 #3622 #3647 #3650

Screenshot 2024-01-18 at 9 58 30 PM

Screenshot 2024-01-18 at 9 58 47 PM

### CODE REVIEW

File: authui/src/phoneInput.ts [hunk 1]
```diff
@@ -13,74 +18,132 @@ interface PhoneInputCountry {
...
-      })
-      .sort((a, b) => a.name.localeCompare(b.name));
+    this._countries = countryCodes.map((countryCode) => {
+      const countryLocalizedName = territories[countryCode];
+      const countryName =
+        defaultTerritories.main.en.localeDisplayNames.territories[countryCode];
+      const countryFlag = getEmojiFlag(countryCode);
+      const countryCallingCode = getCountryCallingCode(countryCode);
+      return {
+        flagEmoji: `<span class="country-flag-icon phone-input__country-flag">${countryFlag}</span>`,
```
Comment (IniZio) [inline authui/src/phoneInput.ts:140]: Need to add to tailwind safelist?

Comment (tung2744) [reply authui/src/phoneInput.ts:140]: I included `"./src/**/*.{css,scss,ts}"` in tailwind content config, so I think it should catch call classes in typescript files. (btw this line has no tailwind classes)

File: resources/authgear/templates/en/translation.json [hunk 2]
```diff
@@ -929,10 +929,15 @@
   "v2-sign-up.signup-with-username": "Continue with Username",
   "v2-sign-up.signup-with-phone": "Continue with Phone",
 
+  "v2-login.title": "Sign in to {AppOrClientName}",
+  "v2-login.subtitle": "It''s free. No credit card needed.",
+  "v2-login.switch-to-sign-up": "No account? <a href=\"{href}\">Create one</a>",
```
[the remark this answers is not in the document]
Comment (tung2744) [reply resources/authgear/templates/en/translation.json:934]: Currently, body-text will style all `<a>` children elements so class is not needed. We can add a class for it if you prefer.

[the remark this answers is not in the document]
Comment (louischan-oursky) [reply resources/authgear/templates/en/translation.json:934]: No need. Let's stick with what we have now.

File: authui/src/customSelect.ts [hunk 3]
```diff
@@ -81,17 +86,22 @@ export class CustomSelectController extends Controller {
...
     this._computePositionCleanup();
 
-    this.triggerTarget.addEventListener("keydown", this.handleKeyDown);
+    this.triggerTarget.removeEventListener("keydown", this.handleKeyDown);
     this.dropdownTarget.removeEventListener("keydown", this.handleKeyDown);
     document.removeEventListener("click", this.handleClickOutside);
   }
 
-  optionValuesChanged() {
+  optionsValueChanged() {
```
[the remark this answers is not in the document]
Comment (tung2744) [reply authui/src/customSelect.ts:101]: It is needed to observe for change of data-custom-select-options

File: authui/src/phoneInput.ts [hunk 4]
```diff
@@ -48,6 +49,7 @@ export class PhoneInputController extends Controller {
       value = `+${country?.phone}${value}`;
     }
     this.inputTarget.value = value;
+    this.inputTarget.dispatchEvent(new Event("input", { bubbles: true }));
```
[the remark this answers is not in the document]
Comment (tung2744) [reply authui/src/phoneInput.ts:52]: Removed in new commits

**Pull request merged**